### ISSUE TITLE
Support for async API calls

### ISSUE THREAD

**Issue by csparpa** [csparpa/pyowm#246]
This would leverage Python 3.5's async/await features

API calls could just be async functions (aka coroutine functions) that pyowm returns and then get executed on an eventloop provided by client code

*note*: asyncio is part of the python std library only from 3.5 so 3.4 doesn't feature it (but asyncio can be imported as an external dependency)

### Design solution
Make PyOWM 2.x sync functions a "sub-case" of async functions as stated here: https://stackoverflow.com/questions/53521716/enhance-synchronous-software-api-to-allow-asynchronous-consuming/53523001#53523001

**Comment by csparpa** [comment]
Take a look here for sync/async library calls handling: https://hackernoon.com/controlling-python-async-creep-ec0a0f4b79ba

**Comment by Harmon758** [comment]
This is something I'd be interested in. Any specific reason it's now wontfix?

**Comment by csparpa** [comment]
@Harmon758 No user interest so far.. and then you shouted ;-) I don't want to rule this feature out if it eventually proves useful !

**Comment by y2kbadbug** [comment]
I'll throw my hat into the game, I am interested as my bot I plan on using OWM with is already async.

**Comment by aitorllj93** [comment]
@csparpa  I personally would appreciate if you mind reopening this

**Comment by csparpa** [comment]
@aitorllj93 @y2kbadbug @Harmon758 open again :)

Are you able to collaborate on this?

**Comment by y2kbadbug** [comment]
So currently I implemented this on my own vanilla script using Python Treq to make the asynchronous calls.  It is written on top of Twisted.

https://pypi.org/project/treq/

This might not be ideal for this project as the level of dependencies it can add, but when I get some free time I'll be happy to poke at a solution for this project.

**Comment by TheBlackPlague** [comment]
I’m not sure “wontfix” is the right label here. This library is a wrapper around network I/O, which is inherently one of the slowest and most blocking kinds of work. Async support seems directly relevant to this use case, and it’s surprising to dismiss it on the basis of “user interest” when the benefit is clear: avoiding unnecessary blocking in applications that already use event loops.

P.S. This is even more important for Python >= 3.14

**Comment by csparpa** [comment]
@TheBlackPlague here you go. Would you contribute to this ?

**Comment by TheBlackPlague** [comment]
> [@TheBlackPlague](https://github.com/TheBlackPlague) here you go. Would you contribute to this ?

Yes, of course. Is there a list of API methods we need converted to async? That way I have a gist on where to start.
